king_phisher/client/windows/plugin_manager.py

```diff
@@ -135,6 +168,76 @@ def load_plugins(self):
 				"{0} (Load Failed)".format(name)
 			))
 
+		if not self.catalog_plugins.catalog_ids():
+			return
+		if not self.catalog_plugins.get_cache():
+			return
+
+		if self.catalog_plugins.catalog_ids():
+			for catalogs in self.catalog_plugins.catalog_ids():
+				catalog = store.append(None, (catalogs, True, None, 'Catalog', catalogs, None, None, False, False))
+				for repos in self.catalog_plugins.get_repos(catalogs):
+					repo = store.append(catalog, (repos.id, True, None, 'Repository', repos.title, None, None, False, False))
+					plugin_collections = self.catalog_plugins.get_collection(catalogs, repos.id)
+					client_plugins = list(plugin_collections)
+					client_plugins.sort()
+					for plugins in client_plugins:
+						installed = False
+						enabled = False
+						if plugin_collections[plugins]['name'] in self.config['plugins.installed']:
+							if repos.id == self.config['plugins.installed'][plugin_collections[plugins]['name']][1]:
+								installed = True
+								enabled = True if plugin_collections[plugins]['name'] in self.config['plugins.enabled'] else False
```

Comment:
Oh god my eyes... that indentation level... :scream: 

I'm just going to skip reviewing this whole block, but suffice to say it looks like it could stand to be refactored into multiple sub functions.